components/mentions/Mentions.razor.cs

```diff
@@ -285,22 +318,15 @@ private void SetOptionsToShow(IEnumerable<MentionsDynamicOption> optionsToShow)
             ShowOptions.AddRange(optionsToShow);
         }
 
-        private Task ShowOrHideBasedOnAvailableShowOptions()
-        {
-            return ShowOptions.Count > 0
-                ? ShowOverlay(true)
-                : HideOverlay();
-        }
-
         internal async Task ItemClick(string optionValue)
         {
             var focusPosition = await JS.InvokeAsync<int>(JSInteropConstants.GetProp, _overlayTrigger.Ref, "selectionStart");
             var preText = Value.Substring(0, focusPosition);
-            preText = preText.LastIndexOf(Prefix) >= 0 ? Value.Substring(0, preText.LastIndexOf(Prefix)) : preText;
+            preText = preText.LastIndexOf(_currentPrefix) >= 0 ? Value.Substring(0, preText.LastIndexOf(_currentPrefix)) : preText;
             if (preText.EndsWith(' ')) preText = preText.Substring(0, preText.Length - 1);
             var nextText = Value.Substring(focusPosition);
```

Comment:
All `Substring` should be replaced with `AsSpan` so that we can avoid generating a lot of temp strings that lead to fragmented memory allocation.